### ISSUE TITLE
print preview does't work

### ISSUE THREAD

**Issue by joergister** [flathub/org.kde.okular#4]
I start okular from the command line using flatpak run -v org.kde.okular
the I open a PDF in okular
When I then click Print Preview, a new window is opened telling me that the print preview component could't be loadet. 
On the commandline I get the error

`org.kde.okular.ui: Could not find a PS viewer for the preview dialog`

Its probably just not includet as far as I can see from the .json file

I'm on Fedora 28 KDE  running a wayland session
flatpak version 0.11.07
Okular 1.4.1 (Flatpak from flathub)

**Comment by tsdgeos** [comment]
It is included since the PS viewer is okular itself :laughing: 

So either:
 * Okular is not compiled with ps support
 * Okular can't find itself because of how flappak works

Looking at the json seems we're not compiling spectre so i guess that's a problem

**Comment by joergister** [comment]
OK thats in fact a funny problem

**Comment by nedrichards** [comment]
;-) in my defence PS support is marked as optional! Sounds like it shouldn't be. patches very, very welcome.

**Comment by nedrichards** [comment]
Should be fixed by https://github.com/flathub/org.kde.okular/commit/0d61e2b38df4c9580c0e66ccbd52d8b557cdfb67

**Comment by nedrichards** [comment]
This isn't fixed by that for me. (also, my printer isn't visible in the printer dialog). How is the print portal in Qt? Is this something that gets better with 5.11 and would be worth an update?

**Comment by tsdgeos** [comment]
@nedrichards the bug is not about printing, the bug is "print preview doesn't work", does that work now?

**Comment by nedrichards** [comment]
No - sorry if I wasn't clear above. It is not fixed for me. I still get 'could not load print preview part' and:

> org.kde.okular.ui: Could not find a PS viewer for the preview dialog

on the CLI.

**Comment by tsdgeos** [comment]
That's weird i can print preview working just fine with the new version, can you open .ps files?

**Comment by nedrichards** [comment]
Yep, I was able to load https://people.sc.fsu.edu/~jburkardt/data/ps/bell_206.ps in Okular.

**Comment by aleixpol** [comment]
Same here. I could open a PS file but still no Print Preview.

**Comment by tsdgeos** [comment]
I'm very confused, why does it work for me then?

**Comment by aleixpol** [comment]
We sat down with @tsdgeos to look at the issue and we found the problem, we didn't find the solution.
A quick workaround would be running `flatpak run --command=kbuildsycoca5 org.kde.okular`.

The problem is that since we opened okular before the plugin was introduced, the plugin cache database (sycoca, that okular uses) got populated without it. It would have been re-populated but since flatpak doesn't update the timestamp of files (everything dates 1st Jan 1970) the caching code thinks it's already done and doesn't revisit it.

I'd say having flatpak set up the package date on every file would be a good solution for this issue, but this is not something we can fix from okular as far as I understand.
